src/main/java/FreedomPointsCounter.java

```diff
@@ -0,0 +1,120 @@
+import exceptions.InvalidPositionException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Map;
+
+public class FreedomPointsCounter {
+
+    private enum Direction {HORIZONTAL, VERTICAL, DIAGONAL_LEFT, DIAGONAL_RIGHT}
+    private final static int MAX_NUMBER_OF_STONES = 4;
+    @NotNull
+    private Board board;
+    private int whitePlayerScore = 0;
+    private int blackPlayerScore = 0;
+
+    public FreedomPointsCounter(@NotNull Board board) {
+        this.board = board;
+    }
+
+    public int getWhitePlayerScore() {
+        return whitePlayerScore;
+    }
+
+    public int getBlackPlayerScore() {
+        return blackPlayerScore;
+    }
+
+    public void setBoard(@NotNull Board board) {
+        this.board = board;
+    }
+
+    public void count() {
+        this.blackPlayerScore = 0;
+        this.whitePlayerScore = 0;
+        for (Map.Entry<Position, Cell> entry : board) {
```

Comment:
No need to iterate through entries, it's sufficient to consider the positions only

Reply:
I use the iterator that we have made.